cmd/main.go

```diff
@@ -60,12 +61,13 @@ required arguments:
 			Port:              *port,
 			Debug:             *debug,
 		},
-		qbit,
-		http.DefaultClient)
+		qbit)
 
 	if err != nil {
 		panic(err)
 	}
+
+	go printer.PrintState(r)
```

Comment:
I generally advice against a "fire and forget" goroutine unless you really don't care what happens to it. Here's a quick [read](https://dave.cheney.net/2016/12/22/never-start-a-goroutine-without-knowing-how-it-will-stop) from Dave Cheney that is worth looking over.

Global vars and shared state are also generally frowned upon. We should think through how to do this print without running a background loop forever

Reply:
Since this goroutine is just for printing the state information to stdout, i thought that it was not a big deal to fire and forget in this case. (ins't when the program terminate all goroutines terminates?:thinking:). But Even with my uncertainty i'm still exiting the goroutine  using the following approach;

1. StopPrintCH inside the ratiospoof state struct
2. my printer.go func listen this channel and exits if necessary 
3. I detect the program shutdown this way: https://github.com/ap-pauloafonso/ratio-spoof/blob/804fe5c0de0d0f8537cd7bde7317cc45b0b63831/internal/ratiospoof/ratiospoof.go#L117-L135

Reply:
You bring up some good points. This is a good use of channels and the printer appears to be shutting down cleanly. So it's not a bad implementation. That said, I try use errgroup or waitgroups over channels and direct goroutine calls  where possible.

In the interest of keeping this PR fairly small, let's save the refactor of the `generate-announce->wait internal->fire announce -> repeat` loop and perhaps the printer for a new PR.